apps/api/src/app/subscribers-v2/usecases/get-subscriber-preferences/get-subscriber-preferences.usecase.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@nestjs/common';
+import {
+  GetSubscriberGlobalPreference,
+  GetSubscriberGlobalPreferenceCommand,
+  GetSubscriberPreference,
+  GetSubscriberPreferenceCommand,
+} from '@novu/application-generic';
+import { ISubscriberPreferenceResponse } from '@novu/shared';
+import { plainToInstance } from 'class-transformer';
+import { GetSubscriberPreferencesCommand } from './get-subscriber-preferences.command';
+import {
+  GetSubscriberPreferencesDto,
+  GlobalPreferenceDto,
+  WorkflowPreferenceDto,
+} from '../../dtos/get-subscriber-preferences.dto';
+
+@Injectable()
+export class GetSubscriberPreferences {
+  constructor(
+    private getSubscriberGlobalPreference: GetSubscriberGlobalPreference,
+    private getSubscriberPreference: GetSubscriberPreference
+  ) {}
+
+  async execute(command: GetSubscriberPreferencesCommand): Promise<GetSubscriberPreferencesDto> {
+    const globalPreference = await this.fetchGlobalPreference(command);
+    const workflowPreferences = await this.fetchWorkflowPreferences(command);
+
+    return plainToInstance(GetSubscriberPreferencesDto, {
+      global: globalPreference,
+      workflows: workflowPreferences,
+    });
```

Comment:
I think we should serialize response to DTO like this in all new endpoints, maybe better in controller instead of usecase - it ensures response is validated and transformed to what is specified in DTO class.

It can be also setup globally on all controlles so this line is not needed.
https://docs.nestjs.com/techniques/serialization#overview

Reply:
I'm not sure if that is needed because everything that comes from the DAL layer should already be valid. Otherwise, it means that we have a corrupted state in DB. 
But I would like to hear some other opinions.

BTW `plainToInstance` can be done on the base class that we can extend from, so we don't need to do it every time.

Reply:
What I mean is the DTO that comes as a response from API, DAL layer is business logic, this is to better illustrate - we dont serialize and we think we do, in order for ClassSerializer decorator to work, you need to return instance from a controller (or it can be also automatized by other decorator) - but we dont do it correctly.

![image](https://github.com/user-attachments/assets/9b7f4461-2915-4387-92a4-71222bd71cc5)